component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPTokenGenerator.java

```diff
@@ -244,4 +261,34 @@ private static void sendNotification(String tenantAwareUsername, String firstNam
 					"MAILTO transport sender is not defined in axis2 configuration file");
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Method to Trigger the TOTP otp event.
+	 *
+	 * @param userName : Identity user name
+	 * @param tenantDomain : Tenant domain of the user.
+	 * @param userStoreDomainName : User store domain name of the user.
+	 * @param notificationEvent : The name of the event.
+	 * @param otpCode : The OTP code returned for the authentication request.
+	 * @throws AuthenticationFailedException : In occasions of failing sending the email to the user.
+	 */
+	private static void triggerEvent(String userName, String tenantDomain, String userStoreDomainName,
+			String notificationEvent, String otpCode) throws AuthenticationFailedException {
+
+		String eventName = IdentityEventConstants.Event.TRIGGER_NOTIFICATION;
+		HashMap<String, Object> properties = new HashMap<>();
+		properties.put(IdentityEventConstants.EventProperty.USER_NAME, userName);
+		properties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, userStoreDomainName);
+		properties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, tenantDomain);
+		properties.put(TOKEN, otpCode);
+		properties.put(TOTPAuthenticatorConstants.TEMPLATE_TYPE, notificationEvent);
+		Event identityMgtEvent = new Event(eventName, properties);
+		try {
+			System.out.println(TOTPDataHolder.getInstance().getIdentityEventService());
+			TOTPDataHolder.getInstance().getIdentityEventService().handleEvent(identityMgtEvent);
+		} catch (IdentityEventException e) {
+			String errorMsg = "Error occurred while calling triggerNotification. " + e.getMessage();
```

Comment:
Let's have more context full eror message. 
1.there is no triggerNotification method.
2.dont use camel case inside the log.
3.lets mention which notification type failed.